crates/bin/prove_block/src/utils.rs

```diff
@@ -0,0 +1,43 @@
+use std::collections::HashSet;
+
+use cairo_vm::Felt252;
+use starknet::core::types::{ExecuteInvocation, FunctionInvocation, TransactionTrace, TransactionTraceWithHash};
+
+// Receives the transaction traces of a given block
+// And extract the contracts addresses that where subcalled
+// TODO: check if we can handle this just reexecuting tx using blockifier
+pub(crate) fn get_subcalled_contracts_from_tx_traces(traces: Vec<TransactionTraceWithHash>) -> HashSet<Felt252> {
```

Comment:
Do you really need ownership of `traces`? Looking at the function I feel like you'd be fine taking a `&[TransactionTraceWithHash]` slice, especially since you return a set of felts which are `Copy`.

Reply:
@odesenfans thanks! You are right :) 
Changed